src/test/java/io/github/thebusybiscuit/slimefun4/testing/tests/researches/TestResearchUnlocking.java

```diff
@@ -47,7 +47,7 @@ private Player awaitUnlock(Player player, Research research, boolean instant) th
             latch.countDown();
         });
 
-        latch.await(10, TimeUnit.SECONDS);
+        latch.await(instant ? 1 : 10, TimeUnit.SECONDS);
```

Comment:
this doesn't really matter, this is just a max wait time. As soon as the latch hits zero this will no longer be blocking

Reply:
I added this because the non-instant unlocking process takes longer than 1 second so the test would fail.

Reply:
Exactly why it was 10 seconds, you made it 1 second if instant 🤔

if it was just as it was, this should be fine.

Reply:
I made it 1 second so it 'would fail' if the instant research took longer, is that wrong?

Reply:
Oh I see what you mean... definitely not a great way to do this. It would work but not exactly ideal

Reply:
Reverted to the previous version